Improve mapping configuration parser

vtorhonen opened the pull request:
Issue #55 shows that the mapping config is not parsed in full if a terminating newline is missing. The last metric mapping is not parsed at all and no errors are shown. This PR changes the parsing method by treating the last line of the file as if it would be a newline between two mappings.

Added two new testcases to `mapper_test.go`:

- a single mapping configuration without a terminating newline
- two mapping configurations without a terminating newline

Tested by running `go test` and it gives a big thumbs up. In addition I tested this locally by running `prom/statsd-exporter` (Dockerhub) and the fixed version side-by-side with this mapping:

```
*.test.*
name="test"
foo="$1"
bar="$2 <no newline>
```

Results for the official image:

```
$ curl -s 127.0.0.1:9102/metrics | grep ^statsd_exporter_build_info
statsd_exporter_build_info{branch="master",goversion="go1.6.2",revision="e00d9e8",version="0.3.0"} 1
$ (curl -s 127.0.0.1:9102/metrics | grep ^test) || echo ":("
:(
$ echo "foo.test.bar:1|g" >/dev/udp/127.0.0.1/9125
$ (curl -s 127.0.0.1:9102/metrics | grep ^test) || echo ":("
:(
```

Results after applying this PR:

```
$ curl -s 127.0.0.1:9102/metrics | grep ^statsd_exporter_build_info
statsd_exporter_build_info{branch="",goversion="go1.6.4",revision="",version=""} 1
$ (curl -s 127.0.0.1:9102/metrics | grep ^test) || echo ":("
:(
$ echo "foo.test.bar:1|g" >/dev/udp/127.0.0.1/9125
$ (curl -s 127.0.0.1:9102/metrics | grep ^test) || echo ":("
test{bar="bar",foo="foo"} 1
```

Pinging @juliusv as instructed in `CONTRIBUTIONS.md`.

grobie commented:
Instead of adding support to handle files without trailing newlines, I'd rather keep the current format but return an error if it's missing. It's the philosophy of Prometheus to only provide one way to do things.

vtorhonen commented:
Changed the behaviour as you suggested. When a terminating newline is missing an error such as this is shown:

```
time="2017-02-24T09:29:41Z" level=fatal msg="Error loading config:Line 3: missing terminating newline" source="main.go:158"
```

Modified tests and made sure they are working as expected.

While I do disagree with this design it's still better than before (being nothing). Usability-wise the original solution would've been better.

grobie commented:
Thanks a lot for your contribution!

Please follow the git commit message best practices, as described here for example: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
